File: src/BaekJoon/java/baekjoon_1764.java
package BaekJoon.java;
// https://www.acmicpc.net/problem/1764

import java.util.Arrays;
import java.util.LinkedList;
import java.util.Scanner;

public class baekjoon_1764 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int m = sc.nextInt();
		int cnt = 0;
		String[] str1 = new String[n];
		String[] str2 = new String[m];
		LinkedList<String> list = new LinkedList<String>();

		sc.nextLine();

		for(int i = 0; i < n; i++)
			str1[i] = sc.nextLine();

		for(int i = 0; i < m; i++)
			str2[i] = sc.nextLine();

		Arrays.sort(str1);
		Arrays.sort(str2);

		for(int i = 0; i < n; i++) {
			int index = Arrays.binarySearch(str2, str1[i]);

			if(index >= 0) {
				cnt++;
				list.add(str2[index]);
			}
		}

		System.out.println(cnt);

		for(int i = 0; i < cnt; i++)
			System.out.println(list.get(i));

		sc.close();
	}
}

File: src/BaekJoon/Rust/baekjoon_1764.rs
use std::io::stdin;

pub fn baekjoon_1764() {
    let n_m: Vec<usize> = stdin().lines().next().unwrap().unwrap()
        .split_whitespace().map(|s| s.parse::<usize>().unwrap()).collect::<Vec<usize>>();
    let (mut n, mut m) = (n_m[0], n_m[1]);
    let mut str1: Vec<String> = Vec::new();
    let mut str2: Vec<String> = Vec::new();
    
    while n > 0 {
        n -= 1;
        
        let str: String = stdin().lines().next().unwrap().unwrap();
        
        str1.push(str);
    }
    
    while m > 0 {
        m -= 1;
        
        let str: String = stdin().lines().next().unwrap().unwrap();
        
        str2.push(str);
    }
    
    str1.sort();
    str2.sort();
    
    let mut result: Vec<String> = Vec::new();
    
    for i in 0..n_m[0] {
        match str2.binary_search(&str1[i]) {
            Ok(index) => {
                result.push(String::from(&str2[index]));
            },
            Err(_) => {}
        }
    }
    
    println!("{}", result.len());
    
    result.iter().for_each(|(value)| println!("{}", value))
}